extensions/amp-carousel/0.1/slidescroll.js

```diff
@@ -287,6 +287,12 @@ export class AmpSlideScroll extends BaseSlides {
   }
 
   /** @override */
+  unlayoutCallback() {
+    this.slideIndex_ = null;
```

Comment:
I do think this is the right thing to do , but i don't understand how this solves your issue. Both the lightbox and imageLightbox do not call `unlayoutCallback()`. 

Do you have a test file that you have tested this on? (You can add this scenario to test/manual/amp-slidescroll.amp.html). Also you might need to add unit test for this (it would be a super small test but useful).